File: .taira-source-c80bf4d85a97/java/iroha_android/src/main/java/org/hyperledger/iroha/android/KeyManagementException.java
package org.hyperledger.iroha.android;

/**
 * Raised when a key provider cannot complete a requested operation.
 *
 * <p>This exception is surfaced to callers so they can decide whether to retry with a different
 * security preference or surface a user prompt. Providers should include actionable messages.
 */
public final class KeyManagementException extends Exception {

  private static final long serialVersionUID = 1L;

  public KeyManagementException(final String message) {
    super(message);
  }

  public KeyManagementException(final String message, final Throwable cause) {
    super(message, cause);
  }
}

File: kotlin/core-jvm/src/main/java/org/hyperledger/iroha/sdk/crypto/KeyManagementException.kt
package org.hyperledger.iroha.sdk.crypto

/**
 * Raised when a key provider cannot complete a requested operation.
 *
 * This exception is surfaced to callers so they can decide whether to retry with a different
 * security preference or surface a user prompt. Providers should include actionable messages.
 */
class KeyManagementException : Exception {
    constructor(message: String) : super(message)
    constructor(message: String, cause: Throwable) : super(message, cause)
}
